src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1727,79 +1716,70 @@ boolean doStartThread(Runnable runnable) throws RejectedExecutionException {
 
     private int tryExecute(final Runnable runnable) {
         QNode tailNext;
+        PoolThreadNode toUnpark = null;
```

Comment:
This allows us to effectively return both the result `int` and a `PoolThreadNode` to be unparked outside of the lock.